subsys/bluetooth/mesh/adv.c

```diff
@@ -41,8 +41,15 @@ const uint8_t bt_mesh_adv_type[BT_MESH_ADV_TYPES] = {
 
 K_FIFO_DEFINE(bt_mesh_adv_queue);
 
+static void adv_buf_destroy(struct net_buf *buf)
+{
+	bt_mesh_adv_send_end(0, BT_MESH_ADV(buf));
+
+	net_buf_destroy(buf);
```

Comment:
I'm a little worried about `BT_MESH_ADV(buf)->data` may be modified elsewhere.

Because of `send_end_cb` can be processed by app layer, and app layer may block the current process. If other processes apply for buffer at this time, `cb_data` may be overwrite?

Reply:
Yes, it's not the adv pointer that's passed, it's the content.

Reply:
Yes, it would need to make a copy of the callback and context first, like this, for example:
```c
	struct bt_mesh_adv adv = *BT_MESH_ADV(buf);

	bt_mesh_adv_send_end(0, &adv);

	net_buf_destroy(buf);
```